gutenberg/acquire/metadata.py

```diff
@@ -26,23 +28,148 @@
 from gutenberg._util.os import remove
 
 
-_METADATA_CACHE = local_path(os.path.join('metadata', 'metadata.db'))
-_METADATA_DATABASE_SINGLETON = None
-
-
-@contextlib.contextmanager
-def _download_metadata_archive():
-    """Makes a remote call to the Project Gutenberg servers and downloads the
-    entire Project Gutenberg meta-data catalog. The catalog describes the texts
-    on Project Gutenberg in RDF. The function returns a file-pointer to the
-    catalog.
-
-    """
-    data_url = r'http://www.gutenberg.org/cache/epub/feeds/rdf-files.tar.bz2'
-    with tempfile.NamedTemporaryFile(delete=False) as metadata_archive:
-        shutil.copyfileobj(urllib2.urlopen(data_url), metadata_archive)
-    yield metadata_archive.name
-    remove(metadata_archive.name)
+_GUTENBERG_CATALOG_URL = \
+    r'http://www.gutenberg.org/cache/epub/feeds/rdf-files.tar.bz2'
+
+
+class CacheAlreadyExistsException(Exception):
```

Comment:
Thanks for your input. As of [9f3aff2](https://github.com/c-w/Gutenberg/blob/9f3aff2/gutenberg/_domain_model/exceptions.py) we now have `gutenberg.Error` as the top-level class for all exceptions thrown by the API.